app/dbOperations/subjects.go

```diff
@@ -49,11 +50,11 @@ func GetSubjectByName(cosmos gremcos.Cosmos, name string) (model.Subject, error)
 	return getSubjectFromResponse(res)
 }
 
-func CreateSubject(cosmos gremcos.Cosmos, subject model.Subject) (model.Subject, error) {
+func CreateSubject(cosmos gremcos.Cosmos, subject modelAPI.Subject) (modelStorage.Subject, error) {
 	_, err := GetSubjectByName(cosmos, subject.Name)
 
 	if err == nil {
-		return model.Subject{}, errors.New("There is already a subject with name " + subject.Name)
+		return modelStorage.Subject{}, errors.New("There is already a subject with name " + subject.Name)
```

Comment:
minor: consider working with pointers instead as it's in general less verbose and leaves a lower memory and gc overhead